### ISSUE TITLE
Round trip suggestions

### ISSUE THREAD

**Issue by nrenner** [abrensch/brouter#460]
Automatically generate circular route (loop) suggestions for a given starting point and desired route length.

Potentially with some additional constraints like ascend/difficulty.

I'm collecting links I come accross:
[Round trip suggestions (brouter-web Wiki)](https://github.com/nrenner/brouter-web/wiki/Round-trip-suggestions)

Would go well with (#258)
See also (nrenner/brouter-web#236)

**Comment by EssBee59** [comment]
Very good suggestions!
I am very interested in 2 new functions:
-penalty points for routes with noise (traffic)
-plus points for pleasant routes (green, sights etc..)

**Comment by afischerdev** [comment]
As we are nearly ready with the 'dynamic range' (#747) we could now offer a simple round trip logic:

From the starting point there is a direction and a distance, this will generate two additional points +/-, say 20 degrees from the direction and then routing can continue as normal.

![roundtrip](https://github.com/user-attachments/assets/9982903d-478c-4661-a00b-764703512e0e)

**Comment by devemux86** [comment]
@afischerdev This is very nice!

Do you mean it will require the dynamic range?

And it will generate beelines at the via points?
This may not be desirable to users, as these points are not user-defined.

**Comment by afischerdev** [comment]
@devemux86 
> Do you mean it will require the dynamic range?

Yes, I think so, when point is more then the usual 250 meters away from a way.
But no beeline generation, just find a point.
After first generation a routing client could offer the two via points for moving.

**Comment by Totorrr** [comment]
BRouter engine now has a the round trip mode implemented! 

See also https://github.com/nrenner/brouter-web/issues/236#issuecomment-3371226858

**Comment by devemux86** [comment]
Round trip planning with BRouter (offline) is also available in **[Cruiser](https://github.com/devemux86/cruiser)** app:

<img src="https://github.com/user-attachments/assets/fa4ed5ec-fce5-426d-aaf6-3ab4a7ec7fea" width="50%"/><img src="https://github.com/user-attachments/assets/a6a04eb6-910b-4c53-b8e1-734707329c66" width="50%"/>
